### PR TITLE
test: refactor card, container, col, and row component tests to typescript and eliminate enzyme

### LINKED ISSUE #6156: Switch tests to @testing-library
To future proof our tests for new React releases, we should probably switch away from enzyme.  Need to convert all tests over to RTL and typescript.

Help would be appreciated here.

If you take on one of the test files, please leave a comment here so there's no overlap.

- [x] AccordionButton - @VibhorJaiswal 
- [x] Accordion
- [x] Alert - @Sanchitverma78
- [x] Badge
- [x] BootstrapModalManager
- [x] BreadcrumItem
- [x] Breadcrumb
- [x] ButtonGroup
- [x] Button
- [x] ButtonToolbar
- [x] CardImg
- [x] Card - @jayeclark
- [x] CarouselCaption
- [x] Carousel
- [x] CloseButton
- [x] Collapse
- [x] Col
- [x] Container
- [x] createChainedFunction
- [x] createUtilityClasses
- [x] DropdownButton
- [x] DropdownItem
- [x] DropdownMenu
- [x] Dropdown
- [x] DropdownToggle
- [x] Fade
- [x] Feedback
- [x] FigureCaption
- [x] Figure
- [x] FloatingLabel
- [x] FormCheck
- [x] FormControlFeedback
- [x] FormControl
- [x] FormGroup
- [x] FormLabel
- [x] FormRange
- [x] FormSelect
- [x] Form
- [x] FormText
- [x] helpers
- [x] Image
- [x] InputGroup
- [x] ListGroupItem
- [x] ListGroup
- [x] ModalBody
- [x] ModalFooter
- [x] ModalHeader
- [x] Modal
- [x] ModalTitle
- [x] NavbarBrand
- [x] NavbarOffcanvas 
- [x] Navbar
- [x] NavbarToggle
- [x] NavDropdown
- [x] NavItem
- [x] NavLink
- [x] Nav
- [x] Offcanvas
- [x] Overlay
- [x] OverlayTrigger
- [x] PageItem
- [x] Pagination
- [x] PlaceholderButton
- [x] Placeholder
- [x] Popover
- [x] ProgressBar
- [x] Ratio
- [x] Row
- [x] Spinner
- [x] SplitButton
- [x] Stack
- [x] TabContent
- [x] Table
- [x] Tabs
- [x] ThemeProvider
- [x] ToastBody
- [x] ToastContainer
- [x] ToastHeader
- [x] Toast
- [x] ToggleButtonGroup
- [x] Tooltip
- [x] useOverlayOffset

### PR DESCRIPTION
**Pull request by jayeclark**
Signed-off-by: Jay Clark <jay@jayeclark.dev>

Partially fixes #6156 

Opening a draft PR to check that I'm on the right path, but I'd happily add additional commits covering more tests. (I could use the practice! 🙂)

### CODE REVIEW

File: test/CardSpec.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import { render } from '@testing-library/react';
+
+import Card from '../src/Card';
+
+describe('<Card>', () => {
+  it('should output a div', () => {
+    const { getAllByText } = render(<Card>Card</Card>);
+    const results = getAllByText('Card');
```
Comment (kyletsang) [inline test/CardSpec.tsx:8]: We can just do a `getByText` here and not worry about checking for multiples

File: test/CardSpec.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  it('allows for the body shorthand', () => {
+    const { getByText } = render(
+      <Card body>test</Card>
+    )
+    getByText('test').classList.contains('card-body').should.be.true;
+  });
+
+  it('Should have div as default component', () => {
+    const { getAllByTestId, getByTestId } = render(<Card data-testid="default-test" />);
+    getAllByTestId('default-test').length.should.equal(1)
```
Comment (kyletsang) [inline test/CardSpec.tsx:68]: We don't need the `getAllByTestId` line here

**Review by kyletsang — CHANGES_REQUESTED** [review]
Few comments, but everything else looks good!

**Comment by jayeclark** [comment]
Great, thanks! I took care of migrating Container, Col, and Row too.

**Review by kyletsang — APPROVED** [review]

**Comment by kyletsang** [comment]
Thanks!

**Pull request merged**